dotnet-desktop-guide/net/wpf/events/snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Media;
+
+namespace CodeSampleCsharp
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml.
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            InitializeComponent();
+        }
+
+        //<ButtonEventHandlers>
+        // The click event handler for the existing button 'ButtonCreatedByXaml'.
+        private void ButtonCreatedByXaml_Click(object sender, RoutedEventArgs e)
+        {
+            // Create a new button.
+            Button ButtonCreatedByCode = new();
+
+            // Specify the button text and background color.
+            ButtonCreatedByCode.Content = "New button and event handler created in code";
+            ButtonCreatedByCode.Background = Brushes.Yellow;
+
+            // Add the new button to the StackPanel.
+            stackPanel.Children.Add(ButtonCreatedByCode);
+
+            // Assign an event handler to the new button using the '+=' operator.
+            ButtonCreatedByCode.Click += new RoutedEventHandler(ButtonCreatedByCode_Click);
+
+            // Assign an event handler to the new button using the AddHandler method.
+            // AddHandler(ButtonBase.ClickEvent, new RoutedEventHandler(ButtonCreatedByCode_Click);
+
+            // Assign an event handler to the StackPanel using the AddHandler method.
+            // stackPanel.AddHandler(ButtonBase.ClickEvent, new RoutedEventHandler(ButtonCreatedByCode_Click));
```

Comment:
I'm not seeing what the point of this commented code is for (just the `stackPanel` one) in both the VB and C#?

Reply:
Fixed, by uncommenting and explaining how a routed event can repeatedly trigger a common event handler as it traverses the element tree.